src/core/layertreemodel.cpp

```diff
@@ -102,6 +102,22 @@ QVariant LayerTreeModel::data( const QModelIndex &index, int role ) const
         return node->isVisible();
       }
     }
+
+    case Traceable:
+    {
+      QgsLayerTreeNode *node = mLayerTreeModel->index2node( mapToSource( index ) );
+      if ( QgsLayerTree::isLayer( node ) )
+      {
+        QgsLayerTreeLayer *nodeLayer = QgsLayerTree::toLayer( node );
+        if ( nodeLayer->layer()->type() == QgsMapLayerType::VectorLayer )
+        {
+          QgsVectorLayer *layer = qobject_cast<QgsVectorLayer *>( nodeLayer->layer() );
```

Comment:
You can skip the check on 112 and directly check `if ( layer )`

Reply:
I had to insert it because otherwise it leaded to a crash on the OSM layer...

Reply:
But I'll double check..

Reply:
I think what that he is saying you can do:
```
QgsVectorLayer *layer = qobject_cast<QgsVectorLayer *>( QgsLayerTree::toLayer( node ) );
if ( layer )
{ 
  …
```

Reply:
I had it like that.
As far I remeber it crashed on the cast...